src/logger/loggerTaskEx.c

```diff
@@ -52,12 +52,18 @@
 int g_loggingShouldRun;
 int g_configChanged;
 int g_telemetryBackgroundStreaming;
+struct sample * current_sample = NULL;
```

Comment:
So why are you going back to this exactly?  IIRC I removed this because of race conditions this would cause at high sample rates.

Reply:
I thought this was the best way for Lua to reach in and get the current sample. 

Since Lua is reaching in and not the logger task making callbacks, this was the best way I could think of.   

Suggestions on alternatives? Make a copy of it in memory and setting the current sample to that instead?

Reply:
Per meeting, this is an acceptable solution with minimal risk.